include/forcing/Forcing.h

```diff
@@ -90,9 +104,32 @@ class Forcing
         read_forcing_aorc(forcing_file_name);
 
         //Initialize forcing vector index to 0;
-        forcing_vector_index_ptr = 0;
+        forcing_vector_index = 0;
     }
 
+    /**
+     * @brief Checks forcing vector index bounds and adjusts index if out of vector bounds
+     */
+    void check_forcing_vector_index_bounds()
```

Comment:
This works, but it creates an assumption that all forcing vectors are guaranteed to be equally sized (the size of the precip vector).  This is currently a valid assumption, but I'm not sure how strong the guarantee is.  Something to keep an eye one in the future.

Reply:
Good point. I added this todo for that. I am unclear why this line didn't show up here.
https://github.com/NOAA-OWP/ngen/blob/f72b754aff80f0186d0222fdd87784e27c1081b0/include/forcing/Forcing.h#L112

Reply:
Kinda weird.  I'll merge and if it still isn't there we can add it in.